fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/sensor/EnvironmentBasicSensorModule.java

```diff
@@ -16,72 +16,57 @@
 package org.eclipse.mosaic.fed.application.ambassador.simulation.sensor;
 
 import org.eclipse.mosaic.fed.application.ambassador.SimulationKernel;
-import org.eclipse.mosaic.fed.application.app.api.sensor.EnvironmentSensorData;
-import org.eclipse.mosaic.fed.application.app.api.sensor.Sensor;
+import org.eclipse.mosaic.fed.application.app.api.sensor.BasicSensorModule;
 import org.eclipse.mosaic.lib.enums.SensorType;
 import org.eclipse.mosaic.lib.objects.environment.EnvironmentEvent;
 
-import java.util.ArrayList;
-import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
-import java.util.function.Consumer;
 
-public class EnvironmentSensor implements Sensor<EnvironmentSensorData> {
-
-    static class EnvironmentData implements EnvironmentSensorData {
-
-        private final Map<SensorType, EnvironmentEvent> environmentEvents = new HashMap<>();
-
-        @Override
-        public int strengthOf(SensorType sensorType) {
-            EnvironmentEvent event = environmentEvents.get(sensorType);
-            if (event != null && event.from <= SimulationKernel.SimulationKernel.getCurrentSimulationTime()
-                    && event.until >= SimulationKernel.SimulationKernel.getCurrentSimulationTime()) {
-                return event.strength;
-            }
-            return 0;
-        }
-    }
+/**
+ * An implementation of the {@link BasicSensorModule} based on {@link EnvironmentEvent}s
+ * broadcast through {@link org.eclipse.mosaic.interactions.environment.EnvironmentSensorUpdates},
+ * e.g., generated by the mosaic-environment simulator.
+ */
+public class EnvironmentBasicSensorModule implements BasicSensorModule {
 
-    private final EnvironmentData currentData = new EnvironmentData();
-    private final Collection<Consumer<EnvironmentSensorData>> callbacks = new ArrayList<>();
+    private final Map<SensorType, EnvironmentEvent> environmentEvents = new HashMap<>();
 
     private boolean enabled = false;
 
     @Override
     public void enable() {
-        this.enabled = true;
+        enabled = true;
     }
 
     @Override
-    public void disable() {
-        this.enabled = false;
-    }
-
     public boolean isEnabled() {
         return enabled;
     }
 
     @Override
-    public EnvironmentSensorData getSensorData() {
-        return currentData;
+    public void disable() {
+        enabled = false;
     }
 
     @Override
-    public void reactOnSensorDataUpdate(Consumer<EnvironmentSensorData> consumer) {
-        callbacks.add(consumer);
```

Comment:
I removed that at least for this basic sensor, as I noticed it was not implemented correctly and doing it right would require more effort. The callback was only triggered when the vehicle entered the event area, but not when it left it or when the event time has reached its end. We can later offer this method, but right now it would not work as expected and would be out of scope of this re-design.